src/secrets/CommandRemove.ts

```diff
@@ -49,13 +50,30 @@ class CommandDelete extends CommandPolykey {
           options: { nodePath: options.nodePath },
           logger: this.logger.getChild(PolykeyClient.name),
         });
-        await binUtils.retryAuthentication(async (auth) => {
-          await pkClient.rpcClient.methods.vaultsSecretsRemove({
-            metadata: auth,
-            secretNames: secretPaths,
-            options: { recursive: options.recursive },
-          });
+        const response = await binUtils.retryAuthentication(async (auth) => {
+          const response =
+            await pkClient.rpcClient.methods.vaultsSecretsRemove();
+          await (async () => {
+            const writer = response.writable.getWriter();
+            let first = true;
+            for (const [vault, path] of secretPaths) {
+              await writer.write({
+                nameOrId: vault,
+                secretName: path,
+                metadata: first
+                  ? { ...auth, options: { recursive: options.recursive } }
+                  : undefined,
+              });
+              first = false;
+            }
+            writer.close();
+          })();
+          return response;
         }, meta);
+
+        if (!(await response.output).success) {
+          throw new errors.ErrorPolykeyCLIFileRead('Failed to delete file');
+        }
```

Comment:
I need feedback on this. How do I handle a non-zero success code?

In Polykey, the RPC will either return a true success code, or throw an error. To ignore the return code (which, in this case, is what we can do) and simply await the output like this:

```ts
await response.output
```

Or do I need to properly handle it?

Reply:
If, after checking, the result is not true, then what do I do? What error should I throw? Currently, I'm just throwing a generic error as I don't know what else can fit the bill.

Do I create a new error for situations where the return code was not true, and throw that? Do I throw another, existing error?

Reply:
Right now, the control flow reaching the `await response.output` statement indicates a success. While we *can* test for success to be true, I don't think it makes sense to do that, especially if we need to gracefully handle it by throwing an error, etc.

That can be implemented when needed. For now, a simple `await` is sufficient.